File: JS/Sorting/bubbleSort.js

function bubbleSort(arr) {
  for (let i = arr.length - 1; i >= 0; i--) {
    let swaps = false;
    for (let j = 0; j < i; j++) {
      if (arr[j] > arr[j + 1]) {
        [ arr[j], arr[j + 1] ] = [ arr[j + 1], arr[j] ];
        swaps = true;
      }
      console.log('Inside iteration: ', j)
    }
    console.log('outside iteration: ', i)
    if (!swaps) break;
  }

  return arr;
}

console.log(bubbleSort([1, 2, 3, 4, 7, 8, 6, 9, 10]))

File: CSharp/Sorting/BubbleSort.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp
{
    public class BubbleSort
    {
        public static void Sort(int[] arr)
        {
            for (var i = arr.Length - 1; i >= 0; i--)
            {
                var swaps = false;
                for (var j = 0; j < i - 1; j++)
                {
                    if (arr[j] > arr[j = 1])
                    {
                        Swap(arr, j, j + 1);
                        swaps = true;
                    }
                }
                if (swaps) break;
            }
        }

        public static void Swap(int[] arr, int i, int j)
        {
            if (i == j) return;
            var temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
}
